app/src/main/java/org/simple/clinic/widgets/PatientSearchResultItemView.kt

```diff
@@ -93,6 +107,48 @@ class PatientSearchResultItemView(
     renderPatientPhoneNumber(searchQuery, model)
     renderVisited(model.lastSeen)
     renderLastSeen(model.lastSeen, currentFacilityId)
+    renderIdentifier(model.identifier, searchQuery)
+  }
+
+  private fun getId(identifier: Identifier, searchQuery: String): Id {
+    val id = when (identifier.type) {
+      BpPassport -> BpPassport.shortCode(identifier)
+      BangladeshNationalId -> identifier.value
+      EthiopiaMedicalRecordNumber -> identifier.value
+      IndiaNationalHealthId -> identifier.value
+      is Unknown -> throw IllegalArgumentException("Unknown identifier: $identifier")
+    }
+    val canHighlightIdentifier = id.contains(searchQuery)
```

Comment:
Hmm, how would `contains` work in the scenario where we save characters like hyphens as part of the ID strings, but people may not necessarily add them when searching?

Reply:
It wouldn't work. Even the search query fails, but when we are creating our identifier we are stripping the special characters (at least for the NHID that has been added recently).